src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/VulnerabilityLifetimeScore.java

```diff
@@ -75,28 +62,33 @@ static double factorOf(Duration lifetime) {
    * Initializes a new score.
    */
   VulnerabilityLifetimeScore() {
-    super("How fast vulnerabilities are patched", VULNERABILITIES, PROJECT_START_DATE);
+    super("How fast vulnerabilities are patched", VULNERABILITIES);
   }
 
   @Override
   public ScoreValue calculate(Value... values) {
     Value<Vulnerabilities> vulnerabilities = findValue(values, VULNERABILITIES,
         "Hey! Give me info about vulnerabilities!");
-    Value<Date> projectStartDate = findValue(values, PROJECT_START_DATE,
-        "Hey! Tell me when the project started!");
 
     if (vulnerabilities.isUnknown()) {
-      return scoreValue(Score.MIN, vulnerabilities, projectStartDate);
+      return scoreValue(Score.MIN, vulnerabilities);
     }
 
-    Date now = new Date(System.currentTimeMillis());
-    Date estimatedIntroducedDate = projectStartDate.isUnknown()
-        ? DEFAULT_PROJECT_START_DATE : projectStartDate.get();
+    if (vulnerabilities.get().isEmpty()) {
+      return scoreValue(Score.MAX, vulnerabilities).explain("No vulnerabilities found.");
+    }
 
     double penalty = 0.0;
+    boolean found = false;
     for (Vulnerability entry : vulnerabilities.get().entries()) {
-      Instant introduced = entry.introduced().orElse(estimatedIntroducedDate).toInstant();
-      Instant fixed = entry.fixed().orElse(now).toInstant();
+      if (!entry.introduced().isPresent() || !entry.fixed().isPresent()) {
+        continue;
+      }
+
+      found = true;
```

Comment:
I think this line should come after #L104. What do you think? 
- The reason being. Consider `if()` in #L93 is `true`. Then the `flag` is already `true` for the next loop.

Reply:
It makes sense. The flag should be set when we're sure that the dates are correct.